xds/src/main/java/io/grpc/xds/LeastRequestLoadBalancerProvider.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2021 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import io.grpc.Internal;
+import io.grpc.LoadBalancer;
+import io.grpc.LoadBalancerProvider;
+import io.grpc.NameResolver.ConfigOrError;
+import io.grpc.Status;
+import io.grpc.internal.JsonUtil;
+import io.grpc.xds.LeastRequestLoadBalancer.LeastRequestConfig;
+import java.util.Map;
+
+/**
+ * Provider for the "least_request_experimental" balancing policy.
+ */
+@Internal
+public final class LeastRequestLoadBalancerProvider extends LoadBalancerProvider {
+  // Minimum number of choices allowed.
+  static final int MIN_CHOICE_COUNT = 2;
+  // Maximum number of choices allowed.
+  static final int MAX_CHOICE_COUNT = 10;
+  // Same as ClientXdsClient.DEFAULT_LEAST_REQUEST_CHOICE_COUNT
+  @VisibleForTesting
+  static final Integer DEFAULT_CHOICE_COUNT = 2;
+
+  // Use GRPC_EXPERIMENTAL_ENABLE_LEAST_REQUEST if set,
+  // otherwise use the io.grpc.xds.experimentalEnableLeastRequest system property.
+  private static final boolean enableLeastRequest =
```

Comment:
The flag defined in https://github.com/grpc/grpc-java/pull/8739/files#diff-6b019eb1d3518381d336168847890513a235ea4974fcb3c76a2e9ace43f1d460R168-R171 already controls the flow, and should be the single source of truth. This one seems redundant and `isAvailable()` can just return `true`. If `isAvailable()` returning `false` does get called, the channel will panic.

Reply:
It seems like it would "fallback" to `round_robin` in this case since when `isAvailable()` returns `false` here, `lbRegistry.getProvider("least_request_experimental")` would return `null` and be captured
https://github.com/grpc/grpc-java/blob/4e236e3c99a672d75b7b9f0d91e49348746eb108/xds/src/main/java/io/grpc/xds/CdsLoadBalancer2.java#L194-L201 

Still, I think it makes sense to use the other flag in ClientXdsClient as source of truth so will update!